### PR TITLE
Make the mode a parameter

### PR DESCRIPTION
**Pull request by daisukixci**
#### Pull Request (PR) description
Make the mode for cron.allow/deny files a parameter as compliance framework like CIS ask the file to be 0640 maximum

### CODE REVIEW

**Review by alexjfisher — CHANGES_REQUESTED** [review]
Thanks for the PR!
I wonder if the parameter name should be more specific than just `mode`?
Could you add documentation for the parameter in the same puppet-strings format as the others?
Instead of type `String` you could use `Cron::Mode`. (We should probably start using Stdlib::Filemode TBH).

File: manifests/init.pp [hunk 1]
```diff
@@ -28,6 +28,7 @@
   Array[Cron::User]    $users_deny         = [],
   Boolean              $manage_users_allow = false,
   Boolean              $manage_users_deny  = false,
+  String               $mode               = '0640',
```
Comment (smortex) [inline manifests/init.pp:31]:
```suggestion
  String               $mode               = '0644',
```

Don't change the default :-)

Comment (daisukixci) [reply manifests/init.pp:31]: Even if it's considered more secure ?

**Comment by daisukixci** [comment]
> I wonder if the parameter name should be more specific than just `mode`?

Any suggestions ?

File: manifests/init.pp (hunk 1 above)
Comment (smortex) [reply manifests/init.pp]: Changing the default would be a breaking change, that's probably not necessary: these files contain name of users, not sensitive data, so hiding it's content is basically just security through obscurity, i.e. not security at all according to some :-)

**Review by smortex — APPROVED** [review]

**Comment by kenyon** [comment]
Seems like `Cron::Mode` should be replaced by [Stdlib::Filemode](https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/REFERENCE.md#stdlibfilemode).

**Review by alexjfisher — APPROVED** [review]

**Comment by alexjfisher** [comment]
> Seems like `Cron::Mode` should be replaced by [Stdlib::Filemode](https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/REFERENCE.md#stdlibfilemode).

There are a lot of other instances, so I think this can be a separate PR.

**Comment by alexjfisher** [comment]
@daisukixci Thanks!

**Comment by daisukixci** [comment]
@alexjfisher Any ETA for next version push to the forge ?

**Pull request merged**